Set up GitHub Action to build and publish NPM package

eliot-akira opened the pull request:
This pull request adds a GitHub Action to build and publish the NPM package `@tangible/fields`.

You can see the build step completes successfully, with compiled files in `publish/npm`.

- https://github.com/TangibleInc/fields/actions/runs/16373535297/job/46267648513?pr=46#step:5:1

Related:

- #44 
- #45 

---

@zinigor or @nicolas-jaussaud - Could you please take over this PR and complete the step to actually publish to NPM?

- [GitHub Docs: Publishing Node.js packages](https://docs.github.com/en/actions/how-tos/use-cases-and-examples/publishing-packages/publishing-nodejs-packages)

It will require creating a new user account at NPM, and probably need to request an invitation to join the `@tangible` namespace. Unfortunately it's currently tied to my personal user at NPM, and ideally we can remove that dependency for automated publishing of packages.

---

Also just to mention, the Fields module uses [CalVer](https://calver.org/) (calendar versioning like `2025.7.15`) instead of [SemVer](https://semver.org/) (semantic versioning like `1.0.0`).

eliot-akira commented:
I went ahead and completed the steps to publish to NPM.

- Log in to npmjs.com
- Create granular access token scoped to namespace `@tangible`
- In the repository's Settings -> Security -> Secrets and Variables -> Actions, create a new repository-level secret named `NPM_TOKEN`. This is used in the GitHub Action script added by this pull request.

zinigor commented:
I have created a new account with the name `tangiblebot` that we can use for shipping Tangible software to NPM. Let me know if you would like to change the tokens and add it as a maintainer.

zinigor commented:
Thanks, I'll do that!

I don't see an email or an invitation in the NPM interface for `tangiblebot` though, can you try again please?

eliot-akira commented:
> https://www.npmjs.com/~tangiblebot

Yeah that's the exact user name I'm inviting.

<img width="588" height="252" alt="image" src="https://github.com/user-attachments/assets/94d6740b-0762-4710-9dc8-96260bb5d225" />

https://www.npmjs.com/settings/tangible/members

<img width="466" height="398" alt="image" src="https://github.com/user-attachments/assets/bc3387f2-1a6c-4c3a-a4c4-58d8fe20509b" />

I removed the pending invitation and sent it again.

Maybe we can try entering its email address instead?

eliot-akira commented:
OK, sent.

zinigor commented:
Thanks for sticking with this thing!

I have changed the email address to the same domain, but now it's `npm+tangiblebot@`, and I have successfully received both the alert that the email was changed, and the confirmation link email.

Let's try that address?

eliot-akira commented:
Sure, I sent an invitation to the new email address. That's a hopeful sign that you received a couple notifications from NPM already.

zinigor commented:
You're not going to believe this, but the emails are not there. I have checked spam, I have checked other mailboxes, and I can only see the email change confirmation message, it was the last message the account has received from NPM.

Perhaps you can try adding my personal NPM account? It's the same username as my GitHub username.

eliot-akira commented:
Great, I switched your user to admin and removed the pending invitation for the other email. Good idea to try from your user, thats's an angle we haven't tried.

zinigor commented:
I don't think I can add anyone to the organization as an admin, there are no controls on the page, I can only add packages:
<img width="1909" height="1275" alt="image" src="https://github.com/user-attachments/assets/7a5ee11a-e354-4706-80fe-340364a286be" />

eliot-akira commented:
OK, I changed the user role to "owner", please try again [here](https://www.npmjs.com/settings/tangible/members).
